src/trusted/service_runtime/sel_ldr.c

```diff
@@ -1821,20 +1821,53 @@ void InitializeCage(struct NaClApp *nap, int cage_id) {
 
 /* Find next available fd in cagetable */
 
-int NextFd(int cage_id){
+int CancelFds(struct NaClApp *nap, int userfds[2], int iterations) {
+
+  for (int i = 0; i < iterations; i++) {
+    int naclfd = fd_cage_table[nap->cage_id][userfds[i]];
+    NaClSetDesc(nap, naclfd, NULL);
+    fd_cage_table[nap->cage_id][userfds[i]] = -1;
+  }
+
+}
```

Comment:
When you cancel the fds, you still leave the orphaned descs in rustposix

Reply:
This actually does call lind_close through the inner DescUnref